src/python/pants/engine/scheduler.py

```diff
@@ -536,6 +536,10 @@ def products_request(self, products, subjects):
     throw_root_states = tuple(state for root, state in result.root_products if type(state) is Throw)
     if throw_root_states:
       unique_exceptions = tuple({t.exc for t in throw_root_states})
+
+      if len(unique_exceptions) == 1 and isinstance(unique_exceptions[0], GracefulTerminationException):
```

Comment:
Mmmm... this bit of magic is unpleasant.

It almost makes me wonder whether we shouldn't have a separate/explicit entrypoint for `@console_rule`? Would likely be useful for @wisechengyi's UX PR.

Reply:
Added a TODO for now